students/Tianx/Lesson5/mailroom3.py

```diff
@@ -0,0 +1,189 @@
+
+    Returns:
+        None.
+
+    """
+    if name not in donor_list:
+        added_donor = {name: [donation]}
+        donor_list.update(added_donor)
+    else:
+        donor_list[name].insert(0, donation) # If donor name exists, insert the donation amount rather than updating the dictionary
+
+
+def send_thank_you():
+    """Sending a thank you email using user input.
+
+    Args:
+       None
+
+    Returns:
+       None
+
+    """
+    # Ask user for a donor's name or to display current list of donors, then ask for donation amount
+    print()
+    while True:
+        donor_name = input('======= The Thank you Menu: =======\n'
+                           "Enter 'list' for to see the list of donors\n"
+                           "or enter full name of donor. \n"
+                           "Enter 'exit' to return to the main menu >").title()
+        if donor_name not in dict_of_donors and donor_name != "List" and donor_name != "Exit":
+            print('\"{}\" is not a current donor, adding to the list...'.format(donor_name))
+            response = input('Please enter a donation amount for ' + donor_name + ' >')
+            while True:
+                try:
+                    float(response)
+                    break
+                except ValueError:
+                    response = input('Error: Please enter a number for the donation amount>')
+            donation_amount = response # Convert the amount into a number
+            adding_donor_info(donor_name, donation_amount, dict_of_donors)
+            create_email(donor_name, donation_amount)
+        elif donor_name == "List": #If the user types list show them a list of the donor names and re-prompt.
+            show_donor_list(dict_of_donors)
+        elif donor_name == "Exit": #If the user types exist return to main menu.
+            break
+        elif donor_name in dict_of_donors: #If user enters exsiting donor
+            response = input('Please enter a donation amount for ' + donor_name + ' >')
+            while True:
+                try:
+                    float(response)
+                    break
+                except ValueError:
+                    response = input('Error: Please enter a number for the donation amount>')
+            donation_amount = response
+            adding_donor_info(donor_name, donation_amount, dict_of_donors)
+            create_email(donor_name, donation_amount)
+            break
+
+
+def create_email(name, donation):
+    print('=======Email Template=======')
+    print('Dear {},\n\nThank you for your generosity, your donation of ${:.2f} will be put to good use.\n\n'
+      'Warm regards,\nMailroom Staff'.format(name, float(donation)))
+
+
+def send_all():
+    """Writing a letter for each donor and save them into a directory with the donor's name.
+
+    Args:
+       None
+
+    Returns:
+       None.
+   """
+    strYesNo = input('Save a Thank you letter for all donors? [Y/N] ').strip().lower()
```

Comment:
make sure to following pep8 guidelines, var names should be snake_case -> `yes_no_str`